connectors/src/resources/connector_resource.ts

```diff
@@ -207,6 +207,10 @@ export class ConnectorResource extends BaseResource<ConnectorModel> {
     return this.errorType === "third_party_internal_error";
   }
 
+  get providerConfiguration(): ConnectorProviderConfigurationResource | null {
```

Comment:
You should be able to type with the mapping in the strategy file. We def don't want this generic type here since this is meant to be used in each separate connectors :+1:

Reply:
I had looked into that before submitting. I don't think it can be done easily. 

At some point, we get a connector by id, and that's runtime, so the connector object can't be typed as a XXXConnectorResource itself. Thus, at the time i'm not sure how its provider configuration can be typed as well

Going to have a second look